src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -259,46 +261,41 @@ public LocalRepository localRepositoryFor(GitHubProject project) throws IOExcept
   private LocalRepository loadLocalRepositoryFor(GitHubProject project) throws IOException {
     Objects.requireNonNull(project, "On no! Project is null!");
 
-    synchronized (LOCAL_REPOSITORIES_INFO_LOCK) {
-      LocalRepositoryInfo info = localRepositoriesInfo.get(project.url());
-      if (info == null) {
-        Path repositoryPath = base
-            .resolve(project.organization().name())
-            .resolve(project.name());
-
-        info = new LocalRepositoryInfo(
-            repositoryPath, Date.from(Instant.now()), project.url());
-      }
-
-      if (Files.isRegularFile(info.path())) {
-        LOGGER.warn("{} is a file but it should be a directory, let's remove it", info.path());
-        Files.delete(info.path());
-      }
+    LocalRepositoryInfo info = LOCAL_REPOSITORIES_INFO.get(project.url());
```

Comment:
All access to `LOCAL_REPOSITORIES_INFO` should be synchronized. Please don't remove the code for obtaining the lock.

I think that we can just use a monitor of `LOCAL_REPOSITORIES_INFO` to synchronize the access:

```java
synchronized (LOCAL_REPOSITORIES_INFO) {
...
}
```

Reply:
I think this would be better. Since `LOCAL_REPOSITORIES_INFO` is `static final`. Only one instance will be available across multiple instances of fetcher. Hence, we can use `synchronized(LOCAL_REPOSITORIES_INFO)`. Instead of having a separate lock object `synchronized(LOCAL_REPOSITORIES_INFO_LOCK)`.

@artem-smotrakov @mibo